### PR TITLE
NCL-5468 Add pluginOverride. Refactor code.

### PR DESCRIPTION
**Pull request by rnc**

### CODE REVIEW

**Comment by rnc** [comment]
@dwalluck  Hi. Was looking at the code to see if its possible to add `pluginOverride` (like `dependencyOverride` ). Have a look - its still WIP (no tests yet) but it seems to work.

File: common/src/main/java/org/commonjava/maven/ext/common/model/ArtifactPluginWrapper.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Simple wrapper to allow a Plugin to be treated as an ArtifactRef which allows for more generic code.
+ */
+public class ArtifactPluginWrapper extends SimpleArtifactRef
+{
+    @Getter
+    private final Plugin original;
+
+    public ArtifactPluginWrapper( Plugin p )
+    {
+        super (p.getGroupId(), p.getArtifactId(), p.getVersion(), null, null);
```
Comment (dwalluck) [inline common/src/main/java/org/commonjava/maven/ext/common/model/ArtifactPluginWrapper.java]: Should type be set to `"maven-plugin"`?

File: core/src/main/java/org/commonjava/maven/ext/core/groovy/ASTTransformer.java [hunk 2]
```diff
@@ -194,18 +206,11 @@ private void changeBaseScriptType(final AnnotatedNode parent, final ClassNode cN
 
         List<AnnotationNode> annotations = parent.getAnnotations( DEPRECATED_COMMAND_TYPE );
         if (cNode.getAnnotations( DEPRECATED_COMMAND_TYPE ).isEmpty()) { // #388 prevent "Duplicate annotation for class" AnnotationFormatError
-
-            // TODO : handle scenario where the annotation is not fully qualified which comes through as a variable
```
Comment (dwalluck) [inline core/src/main/java/org/commonjava/maven/ext/core/groovy/ASTTransformer.java]: Anything added for these?

File: integration-test/src/main/java/org/commonjava/maven/ext/integrationtest/invoker/DefaultExecutionParser.java [hunk 3]
```diff
@@ -40,12 +40,7 @@
         add( ExecutionParser.SYSTEM_PROPERTIES_HANDLER );
     }};
 
-    private List<ExecutionParserHandler> handlers;
-
-    public DefaultExecutionParser()
-    {
-        this( new ArrayList<ExecutionParserHandler>() );
-    }
+    private final List<ExecutionParserHandler> handlers;
```
Comment (dwalluck) [inline integration-test/src/main/java/org/commonjava/maven/ext/integrationtest/invoker/DefaultExecutionParser.java]: Would it make sense to use `Collections.unmodifiableList()` here, too?

File: core/src/main/java/org/commonjava/maven/ext/core/groovy/ASTTransformer.java (hunk 2 above)
Comment (rnc) [reply core/src/main/java/org/commonjava/maven/ext/core/groovy/ASTTransformer.java]: No, this code came from the original classes, so its just cleaning it up

File: common/src/main/java/org/commonjava/maven/ext/common/model/ArtifactPluginWrapper.java (hunk 1 above)
Comment (rnc) [reply common/src/main/java/org/commonjava/maven/ext/common/model/ArtifactPluginWrapper.java]: No I don't think so. The Atlas type is not referring to the pom packaging type but the 'type' of the artifact e.g https://repo1.maven.org/maven2/org/apache/maven/plugins/maven-compiler-plugin/3.8.1/ is a jar.

File: integration-test/src/main/java/org/commonjava/maven/ext/integrationtest/invoker/DefaultExecutionParser.java (hunk 3 above)
Comment (rnc) [reply integration-test/src/main/java/org/commonjava/maven/ext/integrationtest/invoker/DefaultExecutionParser.java]: I've taken another look and simplified the code more. It now implicitly uses a fixed size list.

**Pull request merged**